src/OmniSharp.Script/ScriptProjectSystem.cs

```diff
@@ -107,28 +131,43 @@ public void Initalize(IConfiguration configuration)
                     inheritedCompileLibraries.AddRange(DependencyContext.Default.CompileLibraries.Where(x =>
                             x.Name.ToLowerInvariant().StartsWith("system.runtime")));
                 }
-
             }
 
             // inject all inherited assemblies
             foreach (var inheritedCompileLib in inheritedCompileLibraries.SelectMany(x => x.ResolveReferencePaths()))
             {
-                Logger.LogDebug("Adding implicit reference: " + inheritedCompileLib);
-                AddMetadataReference(Context.CommonReferences, inheritedCompileLib);
+                _logger.LogDebug("Adding implicit reference: " + inheritedCompileLib);
+                AddMetadataReference(commonReferences, inheritedCompileLib);
             }
 
-            // Process each .CSX file
+            // Each .CSX file becomes an entry point for it's own project
+            // Every #loaded file will be part of the project too
             foreach (var csxPath in allCsxFiles)
             {
                 try
                 {
-                    CreateCsxProject(csxPath);
+                    var csxFileName = Path.GetFileName(csxPath);
+                    var project = ProjectInfo.Create(
+                        id: ProjectId.CreateNewId(Guid.NewGuid().ToString()),
+                        version: VersionStamp.Create(),
+                        name: csxFileName,
+                        assemblyName: $"{csxFileName}.dll",
+                        language: LanguageNames.CSharp,
+                        compilationOptions: CompilationOptions,
+                        metadataReferences: commonReferences,
+                        parseOptions: ParseOptions,
+                        isSubmission: true,
+                        hostObjectType: typeof(InteractiveScriptGlobals));
+
+                    // add CSX project to workspace
+                    _workspace.AddProject(project);
+                    _workspace.AddDocument(project.Id, csxPath, SourceCodeKind.Script);
+                    _projects[csxPath] = project;
+                    _logger.LogInformation($"Added CSX project '{csxPath}' to the workspace.");
                 }
                 catch (Exception ex)
                 {
-                    Logger.LogError($"{csxPath} will be ignored due to the following error:", ex.ToString());
-                    Logger.LogError(ex.ToString());
-                    Logger.LogError(ex.InnerException?.ToString() ?? "No inner exception.");
+                    _logger.LogError(0, ex, $"{csxPath} will be ignored due to an following error");
```

Comment:
We've got an extension method that allows you to call LogError without needing to pass an event ID now.